### PR TITLE
React3

### PR DESCRIPTION
**Pull request by jcorteza**

### CODE REVIEW

File: src/components/APIContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import GetTimelineButton from "./GetTimelineButton.jsx";
```
Comment (kev-chan) [inline src/components/APIContainer.jsx]: please consider design of where things are located.  Take into account what we've done in previous labs for direction and let's see what you come up with.

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: dist/index.html [hunk 2]
```diff
@@ -3,19 +3,16 @@
...
-    </div>
-    <button id="apiButton" type="button" >Get Twitter Timeline</button>
-    <div id="apiDataContainer">
-        <p>Loading your Twitter timeline...</p>
-    </div>
-    <div id="reactContainer"></div>
+    </header>
+    <div id="root"></div>
+
+    <script crossorigin src="https://unpkg.com/react@16/umd/react.production.min.js"></script>
```
Comment (kev-chan) [inline dist/index.html]: why is this here?

File: src/components/APIContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import TimelineContainer from "./TimelineContainer.jsx";
+import requestHandler from "../js/requestHandler.js";
+
+class APIContainer extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            request: new XMLHttpRequest(),
```
Comment (kev-chan) [inline src/components/APIContainer.jsx]: is this still needed?

File: src/js/requestHandler.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+        request.addEventListener("error", () => {
+            return component.setState({
+                ajaxResponse: [],
+                errorOccured: true
+            });
+        });
+
+        request.addEventListener("readystatechange", () => {
+            if(request.readyState <= 1) {
```
Comment (kev-chan) [inline src/js/requestHandler.js]: use the actual values instead of numerical values

File: src/js/requestHandler.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                return component.setState({processFinished: false})
+            } else if (request.readyState === 4) {
```
Comment (kev-chan) [inline src/js/requestHandler.js]: same as above

File: src/js/requestHandler.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
+const requestHandler = {
+    
+    initRequest(request, component) {
```
Comment (kev-chan) [inline src/js/requestHandler.js]: from a js standpoint this functionally works however does it need to be tied to a "component"?

File: src/js/sendRequest.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+const sendRequest = (request) => {
```
Comment (kev-chan) [inline src/js/sendRequest.js]: is this needed with the requestHandler?

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: dist/index.html (hunk 2 above)
Comment (jcorteza) [reply dist/index.html]: best practice is to put script tags before the closing body tags. That way the browser loads the DOM before loading the scripts. Otherwise conflicts occur when the scripts act on DOM elements that don't exist yet

File: src/components/APIContainer.jsx (hunk 3 above)
Comment (jcorteza) [reply src/components/APIContainer.jsx]: my requestHandler doesn't create the XMLHttpRequest

File: src/js/requestHandler.js (hunk 4 above)
Comment (jcorteza) [reply src/js/requestHandler.js]: readyState return numbers, and request doesn't have a property with a string version of the values

File: dist/index.html (hunk 2 above)
Comment (kev-chan) [inline dist/index.html]: put all script tags in the head please

File: src/components/TimelineContainer.jsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            data: [],
+            errorOccurred: false
+        }
+
+        this.updateStatus = (responseStatus, responseData) => {
+            if(responseStatus === 200) {
```
Comment (kev-chan) [inline src/components/TimelineContainer.jsx]: can you come up with a design so the rendering code doesn't need to know about the status?

File: src/components/TimelineContainer.jsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+                <div id="timelineContainer">
+                    <p>Loading your Twitter timeline...</p>
+                </div>
+            );
+        }
+
+        return (
+            <div id="apiContainer">
+                <button id="apiButton" type="button" onClick={
+                    (e) => {
```
Comment (kev-chan) [inline src/components/TimelineContainer.jsx]: i'd recommend putting the functions into functions and then assigning them in the constructor.  It makes it more readable and easier to search for as well

**Review by kev-chan — CHANGES_REQUESTED** [review]
